server/app/mutations/grid_services/common.rb

```diff
@@ -172,7 +171,7 @@ def common_validations
           end
           string :entrypoint
           array :env do
-            string
+            string matches: /\A\S+(?<!\\)=.*/
```

Comment:
Rebased to split the env validation changes into #2315

That PR just contains a pending spec that requires this PR to pass.